src/backend.py

```diff
@@ -0,0 +1,726 @@
+            print(msg)
+            return msg
+
+        # Get the HMAC from the request headers
+        client_hmac = request.headers.get('X-Auth-HMAC') or request.headers.get('x-auth-hmac')
+        if not client_hmac:
+            msg = json_dumps({"error": "Missing authentication HMAC"}), 401, 'application/json'
+            print(msg)
+            print(request.headers)
+            return msg
+
+        # Retrieve stored password (the key)
+        credentials = ds_read_record("configuration", 0)
+        stored_password = credentials["Gpassword"].encode('utf-8')
+
+        # Construct the message string
+        path = request.path
+        query_string = request.query_string or ""
+        body_str = request.raw_data or ""
+        message_str = current_challenge + path + query_string + body_str
+        message_bytes = message_str.encode('utf-8')
+
+        # Compute expected HMAC
+        expected_digest = hmac_sha256(stored_password, message_bytes)
+        expected_hmac = hexlify(expected_digest).decode()
+
+        if client_hmac == expected_hmac:
+            # Authentication successful
+            current_challenge = None
+            challenge_timestamp = None
+            print("Authentication successful")
+            return handler(request, *args, **kwargs)
+        else:
+            # Authentication failed
+            print("Authentication failed")
+            print(f"Expected HMAC: {expected_hmac}")
+            print(f"Client HMAC: {client_hmac}")
+            print(message_bytes)
+            return json_dumps({"error": "Authentication failed"}), 401, 'application/json'
+
+    return auth_wrapper
+
+
+
+@add_route("/api/auth/challenge")
+def get_challenge(request):
+    global current_challenge, challenge_timestamp
+    # Generate a random nonce (challenge)
+    current_challenge = random_hex(64)
+    challenge_timestamp = time()
+    # Return the nonce to the client
+    return json_dumps({"challenge": current_challenge}), 200, 'application/json'
+
+@add_route("api/auth/password_check", method="POST", auth=True)
+def check_password(request):
+    return "ok", 200
+
+#
+# Static File Server
+#
+for file_path in ls('web'):
+    route = file_path[3:]  # This should give '/index.html' for 'web/index.html'
+    server.add_route(route, create_file_handler(file_path), methods=['GET'])
+    if route == "/index.html":
+        server.add_route("/", create_file_handler(file_path), methods=['GET'])
+
+#
+# Game
+#
+@add_route("/api/game/reboot", auth=True)
+def app_reboot_game(request):              
+    import reset_control
+    reset_control.reset()
+    sleep(2)
+    reset_control.release(True)         
+    server.reset_bootup_counters()
+
+@add_route("/api/game/name")
+def app_game_name(request):
+    import SharedState
+    return SharedState.gdata["GameInfo"]["GameName"], 200
+
+@add_route("/api/game/status")
+def app_gameStatus(request):
+    return game_status_report(request)
```

Comment:
Not currently used in the web UI, should we remove this until we have a more standard method?

Reply:
removed